pkg/controller/pxc/volumes.go

```diff
@@ -181,20 +184,29 @@ func (r *ReconcilePerconaXtraDBCluster) reconcilePersistentVolumes(ctx context.C
 				switch event.Reason {
 				case "Resizing", "ExternalExpanding", "FileSystemResizeRequired":
 					log.Info("PVC resize in progress", "pvc", pvc.Name, "reason", event.Reason, "message", event.Note)
+					pendingResize = true
 				case "FileSystemResizeSuccessful":
 					log.Info("PVC resize completed", "pvc", pvc.Name, "reason", event.Reason, "message", event.Note)
-				case "VolumeResizeFailed":
+				case "VolumeResizeFailed", naming.EventExceededQuota, naming.EventStorageClassNotSupportResize:
 					log.Error(nil, "PVC resize failed", "pvc", pvc.Name, "reason", event.Reason, "message", event.Note)
 
-					if err := r.handlePVCResizeFailure(ctx, cr, configured); err != nil {
-						return err
-					}
-
-					return errors.Errorf("volume resize failed: %s", event.Note)
+					resizeErrors = append(resizeErrors, errors.Errorf("%s pvc resize failed: %s: %s", pvc.Name, event.Reason, event.Note))
+					continue
 				}
 			}
 		}
 
+		if len(resizeErrors) > 0 {
+			if pendingResize {
+				return nil
+			}
```

Comment:
Do we have a specific reason that we are checking if we are in a pending resize state only if resize errors exist? Because if these are not necessarily connected, we can return nil the moment we track a event.Reason related to pending resizing as noted in L.187. 

But maybe I'm missing something,

Reply:
Even if some PVCs failed to resize, we should wait until the resize is complete to print in the logs which PVCs were successfully resized.